### DISCUSSION TITLE
How to do Dependency Injection with the new DSharpPlus.Commands framework?

### DISCUSSION THREAD

**Discussion by killyp** [DSharpPlus/DSharpPlus#2017] [Q&A] [answered]
Is it possible? If so can you point me to an example?

**Comment by VelvetToroyashi** [comment, marked as answer]
I don't have a specific example on hand as I'm on mobile but yes, it is possible and is no different than [normal dependency injection](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection); services are resolved from the IServiceCollection passed to the DiscordClientBuilder, or if going the more integrated route, from the service collection you call .AddDiscordClient to. 

Documentation about the builder is available [here](https://dsharpplus.github.io/DSharpPlus/articles/migration/4x_to_5x.html?tabs=iservicecollection)

**Comment by killyp** [comment]
Perfect, thank you!
